src/check_es_index_exists.py

```
@@ -65,16 +74,19 @@ def main():
 
     if args.user and args.password:
         connect_params['http_auth'] = (args.user, args.password)
-
+    
+    index = args.index
+    if args.daily:
+        index = index + "-" + datetime.now().strftime(args.date_format)
     es = Elasticsearch([args.host], **connect_params)
 
     # suppress print output from es function
     sys.stderr = None
-    if not es.indices.exists(args.index + '*', allow_no_indices=False):
-        print('Index not found in cluster: {}'.format(args.index))
+    if not es.indices.exists(index + '*', allow_no_indices=False):
```

I know it's out of the scope of this change, but I think having the star in here would cause the following situation:

Let's say we have indexes `nginx` and `nginx_project`. The nginx index could be silently dropped and the check for it would never complain, because `nginx*` would match `nginx_project`, no?

---

Afaik this may happen, I would be fine with removing that star. 
I think it was added by @elmatze79 because he wanted to catch the dates behind the index name.

So with this MR we could actually remove the star. Any objections @elmatze79 ?

---

As a quick-fix we could use `index + '-*'` here as we use the dash to separate the date, for our separations we use underscores

---

It would not bring any value to have 
`index-2021.05.18-*` as there is nothing which could come behind it.
Where `index-*` still can cause some trouble, if people don't know about this culprit. 
If there are custom formats e.g. `index-2021.05.18.12` (if someone wants to add the hour for example) then this can be set by using `--date-format`.

So I am still on board with removing the star.

---

You could:
```
    if args.daily:
        index = index + "-" + datetime.now().strftime(args.date_format)
    else:
        index = index + "-*"
```
on line 79 and then remove the star on line 85.

---

For your proposed solution: If the `--daily` flat is used, the `*` is not applied. But if you do not have daily rotated indices, why would you need the star anyway? You know how your index is called then, I would assume.

Keeping it would be cutting corners and would be error prone for users being unaware of the issue.